docs/tutorials/average_optimizers_callback.ipynb

```diff
@@ -0,0 +1,660 @@
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "xHxb-dlhMIzW"
+      },
+      "source": [
+        "## Overview\n",
+        "\n",
+        "This notebook demonstrates how to use Moving Average Optimizer along with the Model Average Checkpoint from tensorflow addons pagkage.\n",
+        "\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "o2UNySlpXkbl",
+        "colab_type": "text"
+      },
+      "source": [
+        "## Moving Averaging \n",
+        "\n",
+        "> The advantage of Moving Averaging is that they are less prone to rampant loss shifts or irregular data representation in the latest batch. It gives a smooothened and a more genral idea of the model training until some point.\n",
+        "\n",
+        "## Stocastic Averaging\n",
+        "\n",
+        "> Stocastic Weight Averaging converges to wider optimas. By doing so, it resembles geometric ensembeling. SWA is a simple method to improve model performance when used as a wrapper around other optimizers and averaging results from different points of trajectory of the inner optimizer.\n",
+        "\n",
+        "## Model Average Checkpoint \n",
+        "\n",
+        "> ```callbacks.ModelCheckpoint``` doesn't give you the option to save moving average weights in the middle of traning, which is why Model Average Optimizers required a custom callback. Using the ```update_weights``` parameter, ```ModelAverageCheckpoint``` allows you to:\n",
+        "1.   Assign the moving average weights to the model, and save them.\n",
+        "2.   Keep the old non-averaged weights, but the saved model uses the average weights."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "MUXex9ctTuDB"
+      },
+      "source": [
+        "## Setup"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "IqR2PQG4ZaZ0",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 34
+        },
+        "outputId": "a683980d-3cf0-4da0-e680-ff6f70364e72"
+      },
+      "source": [
+        "try:\n",
+        "  %tensorflow_version 2.x\n",
+        "except:\n",
+        "  pass\n",
+        "\n",
+        "import tensorflow as tf"
+      ],
+      "execution_count": 2,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "TensorFlow 2.x selected.\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "vc1eVXwjiFle",
+        "colab_type": "code",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 52
+        },
+        "outputId": "111f2d45-8276-4c73-90c7-c3c5a01b7603"
+      },
+      "source": [
+        "!pip install --upgrade tfa-nightly\n",
```

Comment:
Is it necessary to install the nightly or does it work with the 0.8.2 too?

Reply:
Yes, it was redundant. Removed it.

Reply:
`tfds` is also never used. Will remove that too.